src/SectorView.cpp

```diff
@@ -47,6 +47,128 @@ enum DetailSelection {
 static const float ZOOM_SPEED = 15;
 static const float WHEEL_SENSITIVITY = .03f; // Should be a variable in user settings.
 
+#define FFRAC(_x) ((_x)-floor(_x))
+
+enum Side {
+	XMIN,
+	XMAX,
+	YMIN,
+	YMAX,
+	ZMIN,
+	ZMAX,
+	SIDES
+};
+
+using box_t = std::array<int, SIDES>;
+using cache_t = SectorCache::Slave *;
+
+static bool in_box(int x, int y, int z, const box_t &box)
+{
+	return x >= box[XMIN] && x < box[XMAX] &&
+		y >= box[YMIN] && y < box[YMAX] &&
+		z >= box[ZMIN] && z < box[ZMAX];
+}
+
+static int box_max_size(const box_t &box)
+{
+	return std::max(std::max(box[XMAX] - box[XMIN], box[YMAX] - box[YMIN]), box[ZMAX] - box[ZMIN]);
+}
+
+// helper function
+// cache - source of sector and system data
+// crd - coordinates of the origin point in the sector, l.y.
+// originSector - coordinates of the sector containing the origin point
+// targetSector - coordinates of the sector in which we are looking for the nearest system
+// min_dist - here we will accumulate the minimum distance
+// best_sys - here we will accumulate the nearest system
+static void try_sector_for_nearer_system(cache_t cache, const vector3f &crd, const SystemPath &originSector, const SystemPath &targetSector, float &min_dist, SystemPath &best_sys)
+{
+	RefCountedPtr<Sector> ps = cache->GetCached(targetSector);
+	vector3f base(
+		targetSector.sectorX - originSector.sectorX,
+		targetSector.sectorY - originSector.sectorY,
+		targetSector.sectorZ - originSector.sectorZ);
+	base *= Sector::SIZE;
+	if (ps->m_systems.size()) {
```

Comment:
This if statement is redundant, as a for loop operating in the range `[0 .. m_systems.size)` will run zero iterations. The conditional in a for loop is always run before the loop body begins execution, no matter how many iterations have been processed.

Reply:
Done!